src/transformState.js

```diff
@@ -3,9 +3,39 @@
 /**
  * @param {Object} state
  * @param {Object[]} actions
- */
+*/
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeSpecificKeys(action);
+        break;
+
+      case 'clear':
+        removeAllKeys();
```

Comment:
Pass the state object as an argument to the 'removeAllKeys' function, like this: 'removeAllKeys(state)'.